main.go

```diff
@@ -45,4 +55,15 @@ func main() {
 	close(swapIntentOperationBroadcastChan)
 	close(solverOperationSubmitChan)
 	close(governanceSignatureChan)
+
+	// Questions:
+	// entities:
+	//    backend - is this the bundler that gets the intent, distribute to solvers, collect solutio and create the metaTransactions? you can it backend and not bundler as there can be other configurations?
+	//    governance - what is this entity? is it part of the dApp or part of Atlas?
+	//    user - represents a user application or a real user + dApp ui?
+	//
+	// smart contracts - can you make the 9 smart contract to their "owner" - dApp, Atlas, system?
+	// looks like simulator is not in use
+	// what is txBuilder? provided by Atlas or dApp? if by atlas, why a separate contract?
```

Comment:
It's provided by Atlas to simplify building operations. It's a separate contract because it's a utility, and not a core part of Atlas. We also need to break down components into different contracts because the size of smart contracts are limited.

Reply:
see above. Not sure why this is needed. Don't think others (memswap) has something equivalent

Reply:
Atlas doesn't limit its use case to swap intent, like memswap. It's handling a variety of other scenarios, which make the core contract quite big.

Reply:
There are a few differences between Atlas and Memswap, but a big one is that Atlas works with any transaction type that might create mev, whereas Memswap only works with ERC20 or ERC721 swaps as far as I can see.

So, a weird example might be something like buying a share in someone on friend.tech. If solvers for a way to capture mev by backrunning that, that is something that Atlas could be used for. Memswap cannot be used for anything like that because that doesn't fit the narrow case they specialize in: token-to-token swaps.

Because of this general-purpose approach, Atlas needs more code to enable a wider variety of features, and to ensure security around those features. As such, the contract size is bigger.

But from a user and protocol perspective, the Atlas contract being 1 large contract (as it is now) or spread out across a few smaller contracts should not make a difference. The UX and DevX will be the same - all functions that need to be called will be on a single contract.

TxBuilder is purely a tool from our test suite to help build the calldata for Atlas transactions. Atlas can be used without TxBuilder.